pre_commit_hooks/file_contents_sorter.py

```diff
@@ -18,19 +18,19 @@
 
 
 def sort_file_contents(f):
-    before = tuple(f)
-    after = sorted(before)
+    before = list(f)
+    after = sorted([line.strip(b'\n\r') for line in before if line.strip()])
 
     before_string = b''.join(before)
-    after_string = b''.join(after)
+    after_string = b'\n'.join(after) + b'\n'
 
     if before_string == after_string:
         return PASS
-    else:
-        f.seek(0)
-        f.write(after_string)
-        f.truncate()
-        return FAIL
+
```

Comment:
I prefer the lhs here, despite the `else` being superfluous, it is *obvious* at a glance that this code has two disparate branches.